modules/expo-scroll-forwarder/ios/ExpoScrollForwarderView.swift

```diff
@@ -0,0 +1,169 @@
+    lpg.isEnabled = false
+    lpg.delegate = self
+
+    self.cancelGestureRecognizers = [lpg, tg]
+  }
+  
+  // This will be used to cancel the scroll animation whenever we tap inside of the header. We don't need another
+  // recognizer for this one.
+  override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+    self.stopTimer()
+  }
+
+  // This will be used to cancel the animation whenever we press inside of the scroll view. We don't want to change
+  // the scroll view gesture's delegate, so we add an additional recognizer to detect this.
+  @IBAction func callOnPress(_ sender: UITapGestureRecognizer) -> Void {
+    self.stopTimer()
+  }
+  
+  @IBAction func callOnPan(_ sender: UIPanGestureRecognizer) -> Void {
+    guard let rctsv = self.rctScrollView, let sv = rctsv.scrollView else {
+      return
+    }
+
+    let translation = sender.translation(in: self).y
+    let velocity = sender.translation(in: self).y
+
+    if sender.state == .began {
+      self.initialOffset = sv.contentOffset.y
+    }
+
+    if sender.state == .changed {
+      sv.contentOffset.y = self.dampenOffset(-translation + self.initialOffset)
+    }
+
+    if sender.state == .ended {
+      var currentVelocity = sender.velocity(in: self).y
+
+      // A check for a velocity under 250 prevents animations from occurring when they wouldn't in a normal
+      // scroll view
+      if abs(currentVelocity) < 250 {
+        return
+      }
+
+      // Because this is the header, we just need to scroll to the top. We can't be far enough down the screen to where we need
+      // to deal with animating this.
+      if velocity > 0 {
+        self.scrollToOffset(0)
+        return
+      }
+
+      self.enableCancelGestureRecognizers()
+
+      // Clamping the velocity to a maximum of 5000 incase of any weirdness. I haven't seen any cases where we could
+      // easily exceed this number, but just to make sure.
+      currentVelocity = max(currentVelocity, -5000)
+
+      // Ideally, we would use UIView.animate to do this. However, that messes with the FlatList's virtualization.
+      // Running this on a timer instead simulates us actually dragging to update the content offset.
+      var animTranslation = -translation
+      self.animTimer = Timer.scheduledTimer(withTimeInterval: 1.0 / 120, repeats: true) { timer in
+        currentVelocity *= 0.9875
+        animTranslation = (-currentVelocity / 120) + animTranslation
+        sv.contentOffset.y = self.dampenOffset(animTranslation + self.initialOffset)
+
+        if animTranslation <= -100 {
+          self.scrollToOffset(0)
+          self.stopTimer()
+        }
+
+        if abs(currentVelocity) < 5 {
+          self.stopTimer()
+        }
+      }
+    }
+  }
+  
+  func dampenOffset(_ offset: CGFloat) -> CGFloat {
+    // We can probably uncomment this later, but for now let's just not allow refreshing
+    //    if offset < 0 {
+    //      return offset - (offset * 0.55)
+    //    }
+    
```

Comment:
Adding `refreshControl` support to this is pretty janky. I think the move might be to apply a few changes to the actual `RCTRefreshControl` instead. There are a few private methods there that we could use to our advantage if we make them public. This is something that I'd rather do in a separate PR though, so I say let's test the scroll behavior in general before applying patches to RN.

Reply:
What's the consequence in UI if we don't implement this?

Reply:
not great. i'm going to spend some time to see if i can find a good way in this PR.

not having this will prevent pull to refresh behavior which is not great. the problem is that the underlying RCTRefreshControl does some of its own logic (inside of a private method that we can't access here), so there's an overlap in us calling beginRefreshing here and the underlying method calling it as well.

will give this some thought, because i'd like to ship with refreshControl support since it's a pretty crucial thing to have.